### PR TITLE
#171: fix for bad count value from transactional message api

### LINKED ISSUE #171: transactional message API returns wrong `count` value that breaks rest.getBulk
the transactional message API returns a value for "count" that represents the currently returned number of records, instead of the total amount. checking for count != pageSize is a workaround for this

that being said, there might be edge cases in which "total"-count really is equal to pageSize. I would assume that these cases can be handled gracefully because we already handle 0 results.

### PR DESCRIPTION
**Pull request by JoernBerkefeld**
# PR details

## What is the purpose of this pull request? (put an "X" next to an item)

-   [x] Fix
-   [ ] Improvements
-   [ ] New Feature
-   [ ] Documentation updates
-   [ ] Other, please explain:

## What changes did you make? (Give an overview)

- fixes #171

### CODE REVIEW

File: lib/rest.js [hunk 1]
```diff
@@ -70,8 +70,10 @@ module.exports = class Rest {
             }
             if (
                 Array.isArray(collector[iteratorField]) &&
-                collector[iteratorField].length >= temp.count
+                collector[iteratorField].length >= temp.count &&
+                temp.count != temp.pageSize
```
Comment (DougMidgley) [inline lib/rest.js:74]: if pagesize is 25 and there is actually 25 results - not great outcomes

**Review by DougMidgley — CHANGES_REQUESTED** [review]

File: lib/rest.js (hunk 1 above)
Comment (JoernBerkefeld) [reply lib/rest.js:74]: Well, then we have 1 call with 0 results. Give me another option.

It's a workaround, not a fix. Only stmc product can fix it

Comment (DougMidgley) [reply lib/rest.js:74]: I would rather hard code a check which only runs for transactionalmessage then - also put a support ticket id in :) then we can only apply in this one case, even if it is a hard coded exception

**Review by DougMidgley — CHANGES_REQUESTED** [review]

**Comment by JoernBerkefeld** [comment]
@DougMidgley updated according to your idea. Pls review again

File: lib/rest.js (hunk 1 above)
Comment (JoernBerkefeld) [reply lib/rest.js]: updated according to your idea. Pls review again

**Review by DougMidgley — APPROVED** [review]

**Pull request merged**